File: goblinframework-queue/src/main/java/org/goblinframework/queue/api/GoblinQueueProducer.java
package org.goblinframework.queue.api;

import org.goblinframework.queue.QueueSystem;

import java.lang.annotation.*;

@Documented
@Target({ElementType.FIELD, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
@Repeatable(GoblinQueueProducers.class)
public @interface GoblinQueueProducer {

  QueueSystem system() default QueueSystem.KFK;

  String config() default "primary";

  String queue();

  boolean enabled() default true;
}

File: goblinframework-queue/src/main/java/org/goblinframework/queue/producer/GoblinQueueProducer.kt
package org.goblinframework.queue.producer

import org.goblinframework.queue.QueueSystem

@MustBeDocumented
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@Repeatable
annotation class GoblinQueueProducer (

        val queue: String,             // queue name

        val queueSystem: QueueSystem = QueueSystem.KFK,  // queue system type

        val config: String,

        val enable: Boolean = true
)